apps/native-component-list/src/screens/DeviceScreen.tsx

```diff
@@ -0,0 +1,168 @@
+import React from 'react';
+import { ScrollView, View, Text, Button, Platform } from 'react-native';
+import * as Device from 'expo-device';
+import HeadingText from '../components/HeadingText';
+import MonoText from '../components/MonoText';
+
+interface DeviceConstantName {
+  name?: string;
+  value?: any;
+}
+
+interface State {
+  allFeatures?: string;
+  maxMemory?: number;
+  isSideLoadingEnabled?: string;
+  uptime?: number;
+  isRooted?: string;
+  deviceType?: Device.DeviceType;
+}
+
+class DeviceConstants extends React.Component<DeviceConstantName> {
+  render() {
+    let { name, value } = this.props;
+    if (typeof value === 'boolean') {
+      return (
+        <View style={{ marginBottom: 10 }}>
+          <HeadingText>{name}</HeadingText>
+          <MonoText> {String(value)}</MonoText>
+        </View>
+      );
+    } else {
+      return (
+        <View style={{ marginBottom: 10 }}>
+          <HeadingText>{name}</HeadingText>
+          <MonoText> {value}</MonoText>
+        </View>
+      );
+    }
+  }
+}
+
+export default class DeviceScreen extends React.Component<{}, State> {
+  static navigationOptions = {
+    title: 'Device',
+  };
+
+  state = {
+    allFeatures: '',
+    maxMemory: 0,
+    uptime: 0,
```

Comment:
these could be undefined according to `State` typedef. wouldn't it be semantically more correct? 🤔